### ISSUE TITLE
Dagster's sensors don't properly trigger after turning them off then on

### ISSUE THREAD

**Issue by adespotakis** [dagster-io/dagster#3906]
## Summary
When I tried to turn on a turned off sensor, I noticed that it would not execute, no matter how long I waited.

## Reproduction
Given a pipeline with no solids and a brand new sensor, if I go to the UI for the sensor, turn the sensor on, and immediately turn it off, then on again, I get the same problem I described to you guys initially in the general channel.
The pipeline:
```
from dagster import pipeline
@pipeline()
def blank_pipeline():
    pass
```

The sensor:
```
from dagster import RunRequest, SensorExecutionContext, sensor
@sensor(
    pipeline_name="blank_pipeline",
    name="testing2",
    minimum_interval_seconds=30,
)
def test(context: SensorExecutionContext) -> None:
    yield RunRequest(None)
```

The output from dagster-daemon run:
```
2021-03-19 09:40:44 - dagster-daemon - INFO - instance is configured with the following daemons: ['SchedulerDaemon', 'SensorDaemon']
2021-03-19 09:40:44 - SchedulerDaemon - INFO - Not checking for any runs since no schedules have been started.
2021-03-19 09:40:44 - SensorDaemon - INFO - Not checking for any runs since no sensors have been started.
2021-03-19 09:40:49 - SensorDaemon - INFO - Not checking for any runs since no sensors have been started.
2021-03-19 09:40:54 - SensorDaemon - INFO - Checking for new runs for the following sensors: testing2
2021-03-19 09:40:59 - SensorDaemon - INFO - Checking for new runs for the following sensors: testing2
2021-03-19 09:41:04 - SensorDaemon - INFO - Checking for new runs for the following sensors: testing2
...
```

## Dagit UI/UX Issue Screenshots

![image](https://user-images.githubusercontent.com/10254896/111839043-9176f700-88d0-11eb-8e76-3bd372025107.png)

## Additional Info about Your Environment
Dagster version 0.10.7

**Comment by prha** [comment]
This should get resolved with 4d3b896 and released with `0.11.1` late next week.

**Comment by prha** [comment]
This just got released with `0.11.1`
